File: src/ELibrary.Frontend/src/app/modules/shared/domain/dtos/shop/order/orderBookResponse.ts
import { BookResponse, mapBookResponseToBook, OrderBook } from "../../../..";

export interface OrderBookResponse {
    id: string;
    bookAmount: number;
    bookId: number;
    book: BookResponse;
    bookPrice: number;
}

export function mapOrderBookResponseToOrderBook(response: OrderBookResponse): OrderBook {
    return {
        id: response?.id,
        bookAmount: response?.bookAmount,
        bookPrice: response?.bookPrice,
        bookId: response?.bookId,
        book: mapBookResponseToBook(response?.book)
    }
}

File: src/ELibrary.Backend/LibraryShopEntities/Domain/Dtos/Shop/OrderBookResponse.cs
using LibraryShopEntities.Domain.Dtos.Library;

namespace LibraryShopEntities.Domain.Dtos.Shop
{
    public class OrderBookResponse
    {
        public string? Id { get; set; }
        public int BookAmount { get; set; }
        public int BookId { get; set; }
        public decimal BookPrice { get; set; }
        public BookResponse? Book { get; set; }
    }
}
